Makefile

```diff
@@ -0,0 +1,35 @@
+CC := gcc
+CFLAGS := -g -o
+ALL_CFLAGS := -I. $(CFLAGS)
+
+
+LIBS := -lstdc++ -lavcodec -lavformat -lavutil -lswresample -lswscale -lglfw -lGL
+TARGET_DIR := bin/
+
+SRC_PLAYER := main_window.cpp decoder.cpp
+SRC_TCP_PLAYER := tcp_main_window.cpp decoder.cpp tcp_client.c
+SRC_TCP_SERVER := tcp_server.c
+SRC_TRANSCODER := transcoder.c debug.c
+
+.PHONY: all
+
+all: dir player tcp_player tcp_server transcoder
+
+dir:
+	if [ ! -d "$(TARGET_DIR)" ]; then \
+		mkdir -p $(TARGET_DIR); \
+	fi
+
```

Comment:
Currently looks like a patchy work while redundantly using dir variable everywhere. Can we by default make dir the bin folder is created every time the make command is run no matter what binary we want to prepare?

Reply:
I am not getting it, do you know how to? I read in stackoverflow, thats how I did the patchy work.

Reply:
The check you do to see if that particular directory is there or not and to create the directory, can't it be written such a way that those particular set of code executes even if 'make dir' is not provided?

Reply:
> can't it be written such a way that those particular set of code executes even if 'make dir' is not provided?

Well the the dir is made even if we give `make player` and not `make dir` everytime because I put dir as a prerequisite to every other target file in all the rules. I agree it is patchy and redundant putting dir everywhere as a prerequisite, but I haven't come across a better solution. I will need to dig more on that then, but if you know please let me know :)